class-components/src/store/peopleSlice.ts

```diff
@@ -0,0 +1,53 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { Person } from '../helpers/interfaces';
+
+type State = {
+  searchTerm: string;
+  selectedPeople: Person[];
+};
+
+const initialState: State = {
+  searchTerm: '',
+  selectedPeople: [],
+};
+
+const peopleSlice = createSlice({
+  name: 'people',
+  initialState,
+  reducers: {
+    updateSearchTerm: (state, { payload }: { payload: string }) => {
+      return { ...state, searchTerm: payload.trim() };
+    },
+
+    removeSelectedPerson: (state, { payload }: { payload: string }) => {
+      const foundPersonIndex = state.selectedPeople.findIndex(
+        (person) => person.url === payload,
+      );
+
+      if (foundPersonIndex === -1) return { ...state };
+
+      const newPeople = [...state.selectedPeople];
+      newPeople.splice(foundPersonIndex, 1);
+
+      return { ...state, selectedPeople: [...newPeople] };
+    },

```

Comment:
```suggestion
     removeSelectedPerson: (state, { payload }) => {
      state.selectedPeople = state.selectedPeople.filter(
        ({ url }) => url !== payload,
      );
    },
```

 и это все. чет вы с удалением элемента из массива немного усложняете

на счет возвращения стейта типа
`return { ...state, selectedPeople: [...newPeople] };`
в старом редаксе, обычном, без тул кита, редьюсеры так и работают. но когда мы работаем в тулките, то мы как будто мутируем стейт, просто присваивая в него нужные значения. т.е. можно прям пушить в state.selectedPeople. под капотом ниче не мутируется, но работаем мы именно так